plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineStandardWtpProject.java

```diff
@@ -47,9 +50,14 @@ public void execute(IProgressMonitor monitor) throws InvocationTargetException,
     CreateProjectOperation operation = new CreateProjectOperation(
         description, "Creating new App Engine Project");
     try {
-      operation.execute(progress.newChild(50), uiInfoAdapter);
-      
-      CodeTemplates.materialize(newProject, config, progress.newChild(50));
+      operation.execute(progress.newChild(20), uiInfoAdapter);
+      IFacetedProject facetedProject = ProjectFacetsManager.create(
+          newProject, true, progress.newChild(40));
+      IProjectFacet javaFacet = ProjectFacetsManager.getProjectFacet("jst.java");
+      // todo setup sourcepath
+      facetedProject.installProjectFacet(javaFacet.getVersion("1.7"), null, monitor);
+
+      CodeTemplates.materialize(newProject, config, progress.newChild(40));
```

Comment:
It turns out the default source location configured for a `java` facet [can be overridden from a system property](https://git.eclipse.org/c/webtools-common/webtools.common.fproj.git/tree/plugins/org.eclipse.jst.common.project.facet.core/src/org/eclipse/jst/common/project/facet/core/internal/FacetCorePlugin.java#n74].  There's no API for pulling out the source folder for a faceted project, and although you could grovel through the classpath entries of the `IProject`, it's easiest to provide a `JavaFacetInstallConfig` instance that you've explicitly set the source location on via `setSourceFolder(IPath)`.